### DISCUSSION TITLE
How to serializer result like django rest framework

### DISCUSSION THREAD

**Discussion by zhimingzhang123** [fastapi/fastapi#7871] [Questions] [answered]
I have two models:
```python
UserModel:
   name: str
   age: int

ItemsModel:
    desc: str
    user_id: ForeignKey
```
db data:
```python
user = {"id": 1, "name": "user1",  "age": 18}
items = {"id": 1, "desc": "some desc", "user_id": 1}
```
I want the result: (like django rest framework nested serializer)
```python
items: {
  "id": 1, 
  "desc": "some desc", 
  "user": {  # serializer this, do not return user.id 
    "id": 1,
    "name": "user1",
    "age": 18
    }
}
```

**Comment by dmontagu** [comment, marked as answer]
FastAPI doesn't have a built in way to accomplish this -- you'll have to manually create the appropriate "merged" output model.

Something like:

```python
class ItemsMerged(BaseModel):
    desc: str
    user: UserModel

    class Config:
        orm_mode = True
```

That said, by using a [SQLAlchemy Relationship](https://docs.sqlalchemy.org/en/13/orm/relationships.html) it should be possible to return an orm instance in your endpoint function and have the `user` data automatically populate the `user` field of `ItemsMerged`.

As far as I'm aware, there isn't currently a good way to do this asynchronously, or without making use of the SQLAlchemy ORM. If performance is enough of a concern that that's a problem though, I suspect you'd be better off manually writing the relevant query logic anyway.

**Comment by zhimingzhang123** [comment]
thank you, It works very well.

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :clap: :bow: 

Thanks @zhimingzhang123  for reporting back and closing the issue :+1:
